packages/node_modules/@webex/plugin-meetings/src/reconnection-manager/index.js

```diff
@@ -41,6 +56,58 @@ export default class ReconnectionManager {
     this.configure(meeting);
   }
 
+  /**
+   * Sets the iceState to connected and clears any disconnect timeouts and
+   * related timeout data within the iceState.
+   *
+   * @returns {undefined}
+   * @public
+   * @memberof ReconnectionManager
+   */
+  iceReconnected() {
+    if (this.iceState.disconnected) {
+      LoggerProxy.logger.log('ReconnectionManager:index#iceDisconnected --> ice has reconnected');
+
+      this.iceState.resolve();
+      this.iceState.resolve = () => {};
+
+      if (this.iceState.timer && this.iceState.timer.clearTimeout) {
+        this.iceState.timer.clearTimeout();
+      }
+
+      this.iceState.disconnected = false;
+    }
+  }
+
+  /**
+   * Set the iceState to disconnected and generates a timeout that waits for the
+   * iceState to reconnect and then resolves. If the ice state is already
+   * processing a reconnect, it immediately resolves. Rejects if the timeout
+   * duration is reached.
+   *
+   * @returns {Promise<undefined>}
+   * @public
+   * @memberof ReconnectionManager
+   */
+  iceDisconnected() {
+    if (!this.iceState.disconnected) {
+      LoggerProxy.logger.log('ReconnectionManager:index#iceDisconnected --> waiting for ice reconnect');
+
+      this.iceState.disconnected = true;
+
+      return new Promise((resolve, reject) => {
+        this.iceState.timer = setTimeout(() => {
+          reject(new Error(`ice reconnection did not occur in ${this.iceState.timeoutDuration}ms`));
```

Comment:
Let's check `this.iceState.disconnected` one more time here before rejecting. It could be that we didn't clear the timer before it executed, even though we set `disconnected` to false. Probably an edge case but could help with race conditions.

Reply:
I don't know if it is necessary to check `disconnected`. Javascript isn't multithreaded. We should probably set `disconnected` to false like it is in `iceReconnected`

Reply:
I can implement a getter on iceState that retrieves the current state if needed, let me know which is preferred @adamweeks and @mwegman

Reply:
You should add `this.iceState.disconnected = false;` after this line in the Promise so your reconnect logic will keep working after the first timer goes off.

Reply:
Added.